### PR TITLE
support rings in vertex model

### PR DESCRIPTION
**Pull request by 3nids**
with the fill ring tools, it is required to support rings when using the vertex tool

### CODE REVIEW

File: src/core/vertexmodel.cpp [hunk 1]
```diff
@@ -326,39 +335,48 @@ void VertexModel::setCurrentVertex( int newVertex, bool forceUpdate )
...
+VertexModel::Centroid VertexModel::segmentCentroid( int leftIndex, int rightIndex, bool isExtending, bool goingForward )
 {
   Centroid centroid;
 
   QList<int> indexes = QList<int>() << leftIndex << rightIndex;
   std::sort( indexes.begin(), indexes.end() );
 
   if ( indexes[1] - indexes[0] > 1 )
-    indexes[0] = indexes[1] - 1;
+    indexes[0] = indexes[1] - 1;
```
Comment (3nids) [inline src/core/vertexmodel.cpp:346]:
```suggestion
    indexes[0] = indexes[1] - 1;
```

**Comment by m-kuhn** [comment]
Can you add a description?

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-960-2efa9a82d7bda55c3d7223b13eb5309c4e15b817-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-960-2efa9a82d7bda55c3d7223b13eb5309c4e15b817-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-960-2efa9a82d7bda55c3d7223b13eb5309c4e15b817-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-960-2efa9a82d7bda55c3d7223b13eb5309c4e15b817-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-960-30e35a798b5c5045682fde809244c605f5167c84-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-960-30e35a798b5c5045682fde809244c605f5167c84-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-960-30e35a798b5c5045682fde809244c605f5167c84-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-960-30e35a798b5c5045682fde809244c605f5167c84-x86_64.apk)

**Pull request merged**